### PR TITLE
net: networking stack auto configuration

### PR DESCRIPTION
**Pull request by tristanx86-jump**
## Overview
Many (possibly most) validators use the default configuration provided by the default.toml for the networking subsystem. In these cases large amounts of performance are being left on the table since the defaults are designed to be as safe as possible.

Many validators enable features like zero copy on drivers/kernels that are supposed to support it but have a broken driver implementation. This creates unnecessary support burden in Discord when they report issues that are not Firedancer's fault, as well as making the operator have a less smooth experience using Firedancer.

## Info used
This PR lets Firedancer decide on behalf of the user which networking features to enable based on their system's:
- Linux kernel version
- NIC driver
- Whether the networking interface is a bond
- Whether the networking interface is a virtual interface
- How many slaves there are of the bond (if it is a bond)

## Testing
What is deemed safe was mostly decided using the results of fd-linux-version-testbed, (results cannot be attached at this time due to an infosec issue on my mac).

This is not a 100% foolproof way to test as it does not take into account every edge case e.g. each individual Linux minor release revision, however I believe the risk of these edge cases being the reason for a failure to be low (although not 0).

### CODE REVIEW

File: src/app/fdctl/config/default.toml [hunk 1]
```diff
@@ -981,25 +1001,35 @@ dynamic_port_range = "8900-9000"
         # to 100 Gbps per net tile.
         #
         # Only works when XDP is provided by the network driver (see
-        # the xdp_mode option above).  If the kernel/hardware does not
-        # support zero copy, `fdctl run` will fail to start up with
-        # "operation not supported".
-        xdp_zero_copy = false
+        # the xdp_mode option above).  With true chosen on kernel/hardware
+        # which does not support zero copy, `firedancer run` will fail to
+        # start up with "operation not supported".
```
Comment (ripatel-fd) [inline src/app/fdctl/config/default.toml:1006]: Maybe worth mentioning that xdp_zero_copy can be broken even if the driver supposedly supports it. (e.g. ice memory corruption bug)

File: src/app/platform/fd_config_extract.h [hunk 2]
```diff
@@ -111,6 +111,20 @@ fdctl_cfg_get_bool( int *                 out,
...
+fdctl_cfg_get_boolau( int *                 out,
+                      ulong                 out_sz FD_PARAM_UNUSED,
+                      fd_pod_info_t const * info,
+                      char const *          path ) {
+  if( info->val_type==FD_POD_VAL_TYPE_CSTR
+   && !strcmp( info->val, "auto" ) ) {
+    *out = 2;
+    return 1;
+  }
+  return fdctl_cfg_get_bool( out, out_sz, info, path );
```
Comment (ripatel-fd) [inline src/app/platform/fd_config_extract.h:136]: Can you make this accept `"true"` and `"false"` as strings in addition to direct bools so we have a more consistent schema?

File: src/app/shared/fd_config.c [hunk 3]
```diff
@@ -530,11 +533,25 @@ fd_config_validate( fd_config_t const * config ) {
   CFG_HAS_NON_EMPTY( hugetlbfs.max_page_size );
 
   CFG_HAS_NON_ZERO( net.ingress_buffer_size );
+  if( 0!=strcmp( config->net.auto_level, "standard" )
+   && 0!=strcmp( config->net.auto_level, "minimal"  ) ) {
```
Comment (ripatel-fd) [inline src/app/shared/fd_config.c]: nit usually && usually goes on the end of the previous line

File: src/app/shared/fd_config.h [hunk 4]
```diff
@@ -203,16 +205,16 @@ struct fd_config_net {
...
+    int  xdp_zero_copy; /* true/false or "auto" */
+    char poll_mode[ 16 ]; /* "prefbusy", "softirq" or "auto" */
 
     uint xdp_rx_queue_size;
     uint xdp_tx_queue_size;
     uint flush_timeout_micros;
     char rss_queue_mode[ 16 ]; /* "simple", "dedicated", or "auto" */
     int  listen_gre;
-    int  native_bond;
+    int  native_bond; /* true/fase or "auto" */
```
Comment (ripatel-fd) [inline src/app/shared/fd_config.h]:
```suggestion
    int  native_bond; /* true/false or "auto" */
```

**Review by ripatel-fd — DISMISSED** [review]

**Review by ripatel-fd — APPROVED** [review]
